src/util/submit-addon.js

```diff
@@ -0,0 +1,388 @@
+/* @flow */
+import { createWriteStream, promises as fsPromises } from 'fs';
+import { pipeline } from 'stream';
+import { promisify } from 'util';
+
+// eslint-disable-next-line no-shadow
+import fetch, { FormData, fileFromSync, Response } from 'node-fetch';
+import defaultJwt from 'jsonwebtoken';
+
+export type SignResult = {|
+  success: boolean,
+  id: string | null,
+  downloadedFiles: Array<string>,
+|};
+
+type ClientConstructorParams = {|
+  apiKey: string,
+  apiSecret: string,
+  apiUrlPrefix: string,
+  apiJwtExpiresIn?: number,
+  debugLogging?: boolean,
+  validationCheckInterval?: number,
+  validationCheckTimeout?: number,
+  approvalCheckInterval?: number,
+  approvalCheckTimeout?: number,
+  logger?: any,
+  downloadDir?: string,
+|};
+
+export default class Client {
+  apiKey: string;
+  apiSecret: string;
+  apiUrlPrefix: string;
+  apiJwtExpiresIn: number;
+  debugLogging: boolean;
+  validationCheckInterval: number;
+  validationCheckTimeout: number;
+  approvalCheckInterval: number;
+  approvalCheckTimeout: number;
+  logger: any;
+  downloadDir: string;
+
+  constructor({
+    apiKey,
+    apiSecret,
+    apiUrlPrefix,
+    apiJwtExpiresIn = 60 * 5, // 5 minutes
+    debugLogging = false,
+    validationCheckInterval = 1000,
+    validationCheckTimeout = 300000, // 5 minutes.
+    approvalCheckInterval = 1000,
+    approvalCheckTimeout = 900000, // 15 minutes.
+    logger = console,
+    downloadDir = process.cwd(),
+  }: ClientConstructorParams) {
+    this.apiKey = apiKey;
+    this.apiSecret = apiSecret;
+    this.apiUrlPrefix = `${apiUrlPrefix}addons/`;
```

Comment:
tbh I initially did this so all the urls we were building weren't so long - because each one was under `/addons/` I thought it'd be easier to follow if I just added it here.  But it is a bit of hidden functionality and I'm not sure it makes sense.

Reply:
I don't understand the reference to the other review comment, and what the ask is about normalizing `apiUrlPrefix` though - is there something to be done here?